src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,44 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public <V extends Machine> List<V> getAll(Class<? extends Machine> type) {
+        if (Bulldozer.class.equals(type)) {
+            BulldozerProducer bulldozerProducer = new BulldozerProducer();
+            return (List<V>) bulldozerProducer.get();
+        } else if (Excavator.class.equals(type)) {
+            ExcavatorProducer excavatorProducer = new ExcavatorProducer();
+            return (List<V>) excavatorProducer.get();
+        } else if (Truck.class.equals(type)) {
+            TruckProducer truckProducer = new TruckProducer();
+            return (List<V>) truckProducer.get();
+        }
+        return new ArrayList<>();
```

Comment:
This code must be a good example of polymorphism.
Let's create an abstract variable of a type MachineProducer and initialize it with an applicable producer implementation. Then just call the method get one time.

------
(optional) Also, you can try to use a hashMap instead of if-blocks. It is a really good practice.